File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        const int SIZE = nums.size();
        if(SIZE <= 1) return SIZE;
        
        int pos = 0;
        for(int i=0; i<SIZE-1; ++i) {
            if(nums[i] != nums[i+1]) {
                nums[pos++] = nums[i];
            }
        }
        
        if(pos != SIZE-1) nums[pos] = nums[SIZE-1];
        return pos+1;
    }
};

File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.rb
# @param {Integer[]} nums
# @return {Integer}
def remove_duplicates(nums)
    l = 1
    1.upto(nums.size-1) do |r|
        if nums[r] != nums[r-1]
            nums[l] = nums[r]
            l += 1
        end
    end
    
    return l
end